tensorflow/cc/gradients/math_grad_test.cc

```diff
@@ -737,6 +737,108 @@ TEST_F(CWiseUnaryComplexGradTest, Conj) {
   TestCWiseGradComplex(CONJ, x, dy, dx_expected);
 }
 
+class BinaryGradFunctionTest : public ::testing::Test {
+ protected:
+  using TheDataType = float;
+  using CalcExpectedGradient = std::function<TheDataType(
+      const TheDataType x, const TheDataType y, const TheDataType df)>;
+  using CreateFunctionToTest =
+      std::function<Output(const Scope& scope, const Input& x, const Input& y)>;
+
+  void TestGradient(const CreateFunctionToTest& get_function_to_test,
+                    const CalcExpectedGradient& calc_df_dx,
+                    const CalcExpectedGradient& calc_df_dy) {
+    const Scope scope(Scope::NewRootScope().WithDevice("/cpu:0"));
+    const TensorShape tensor_shape(
+        {GetRandForSizes(), GetRandForSizes(), GetRandForSizes()});
+    const Output x = Const(scope, GetRandTensor(tensor_shape));
+    const Output y = Const(scope, GetRandTensor(tensor_shape));
+    const Output df = Const(scope, GetRandTensor(tensor_shape));
+    const Output expected_df_dx = Const(
+        scope, GetExpectedGradient(scope, tensor_shape, x, y, df, calc_df_dx));
+    const Output expected_df_dy = Const(
+        scope, GetExpectedGradient(scope, tensor_shape, x, y, df, calc_df_dy));
+    const Output f = get_function_to_test(scope, x, y);
+
+    // All data & info have been collected => run test itself:
```

Comment:
remove. This should be straightforward from the code and is vulnerable to getting stale after refactorings.